src/Umbraco.Infrastructure/Install/InstallSteps/DatabaseConfigureStep.cs

```diff
@@ -64,8 +65,37 @@ private void ConfigureConnection(DatabaseModel database)
             }
         }
 
+        public override object ViewModel
+        {
+            get
+            {
+                var databases = new List<object>()
+                {
+                    new { name = "Microsoft SQL Server", id = 1 },
+                    new { name = "Microsoft SQL Azure", id = 3 },
+                    new { name = "Custom connection string", id = -1 },
+                };
+
+                if (IsSqlCeAvailable())
+                {
+                    databases.Insert(0,  new { name = "Microsoft SQL Server Compact (SQL CE)", id = 0 });
+                }
+
+                return new
+                {
+                    databases = databases
+                };
+            }
+        }
+
+        public static bool IsSqlCeAvailable()
+        {
+            return !(Type.GetType("Umbraco.Persistance.SqlCe.SqlCeSyntaxProvider, Umbraco.Persistance.SqlCe") is null);
```

Comment:
This may not always work unless the type is loaded into the appdomain already (perhaps it is, ... i'll continue checking in the PR). The alternative way to force it Assembly.Load("....").GetType("....") then you are forcibly loading it into the appdomain.

Reply:
I've left this one but added notes saying that we know we've forcibly loaded this assembly into the appdomain before this is ever called.